src/directives/NameServiceResolver.js

```diff
@@ -0,0 +1,114 @@
+import normalise from '@/helpers/normalise';
+import { Misc } from '@/helpers';
+const NameServiceResolver = {
+  bind: function(el, binding, vnode) {
+    vnode.context.$watch(binding.value, function(e) {
+      const _this = vnode.context;
+      const errorPar = document.createElement('p');
+      errorPar.classList.add('resolver-error');
+      const ens = _this.ens;
+      const removeElements = function() {
+        const child = el.parentNode.parentNode.lastChild;
+        Object.keys(child.classList).forEach(item => {
+          if (child.classList[item] === 'resolver-error') {
+            vnode.elm.parentNode.parentNode.removeChild(child);
+          }
+        });
+      };
+      if (
```

Comment:
I believe this if-else is also unneccesary since `isValidENSorEtherAddress` will verify ETH addresses against the chainID